pkg/strvals/parser.go

```diff
@@ -237,19 +237,19 @@ func (t *parser) key(data map[string]interface{}, nestedNameLevel int) (reterr e
 				_, err = t.emptyVal()
 				return err
 			}
-			//End of key. Consume =, Get value.
+			// End of key. Consume =, Get value.
 			// FIXME: Get value list first
 			vl, e := t.valList()
-			switch e {
-			case nil:
+			switch {
```

Comment:
should probably rewrite this switch as an `if/else if` block now? As effectively each `case` is an `else if` ?

Reply:
(and similar below)